File: client/src/data/entities/tile.ts
export interface Tile {
  id: string,
  guildId: string | undefined,
  task: string,
  description: string,
  conditions: string,
  imagePath: string
}

export interface TileForm {
  task: string,
  description: string,
  conditions: string
}

export interface CreateTileModel {
  guildId: string,
  task: string,
  description: string,
  conditions: string,
  imagePath: string
}

export interface UpdateTileModel {
  tileId: string,
  task: string,
  description: string,
  conditions: string,
  imagePath: string
}

File: API/Entities/Tile.cs
namespace API.Entities;

public class Tile
{
    public required string Id { get; set; }

    public string? GuildId { get; set; }
    public Guild? Guild { get; set; }

    public required string Task { get; set; }
    public required string Description { get; set; }
    public required string Conditions { get; set; }
    public required string ImagePath { get; set; }
}